instrumentation/runtime-metrics/runtime-metrics-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetricsjava8/BufferPools.java

```diff
@@ -46,25 +47,30 @@ public static void registerObservers(OpenTelemetry openTelemetry) {
 
   // Visible for testing
   static void registerObservers(OpenTelemetry openTelemetry, List<BufferPoolMXBean> bufferBeans) {
-    Meter meter = RuntimeMetricsUtil.getMeter(openTelemetry);
+    Meter meter = JmxRuntimeMetricsUtil.getMeter(openTelemetry);
+    observables.add(
+        meter
+            .upDownCounterBuilder("process.runtime.jvm.buffer.usage")
+            .setDescription("Memory that the Java virtual machine is using for this buffer pool")
+            .setUnit("By")
+            .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getMemoryUsed)));
 
-    meter
-        .upDownCounterBuilder("process.runtime.jvm.buffer.usage")
-        .setDescription("Memory that the Java virtual machine is using for this buffer pool")
-        .setUnit("By")
-        .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getMemoryUsed));
-
-    meter
-        .upDownCounterBuilder("process.runtime.jvm.buffer.limit")
-        .setDescription("Total capacity of the buffers in this pool")
-        .setUnit("By")
-        .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getTotalCapacity));
+    observables.add(
+        meter
+            .upDownCounterBuilder("process.runtime.jvm.buffer.limit")
+            .setDescription("Total capacity of the buffers in this pool")
+            .setUnit("By")
+            .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getTotalCapacity)));
+    observables.add(
+        meter
+            .upDownCounterBuilder("process.runtime.jvm.buffer.count")
+            .setDescription("The number of buffers in the pool")
+            .setUnit("{buffers}")
+            .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getCount)));
+  }
 
-    meter
-        .upDownCounterBuilder("process.runtime.jvm.buffer.count")
-        .setDescription("The number of buffers in the pool")
-        .setUnit("{buffers}")
-        .buildWithCallback(callback(bufferBeans, BufferPoolMXBean::getCount));
+  public static void closeObservers() {
```

Comment:
Instead of keeping a global stateful list of observers, WDYT about making the `registerObservers(OpenTelemetry)` function return a `Closeable`? The users can decide how to handle these themselves.

Reply:
I think this is a good idea. I have applied your suggestion.